src/dune/watermarks.ml

```diff
@@ -173,7 +173,7 @@ let subst_file path ~map =
 
 (* Minimal API for dune-project files that makes as little assumption about the
    contents as possible and keeps enough info for editing the file. *)
-module Dune_project = struct
+module Dune_project_mini = struct
```

Comment:
Initially, the intent of this sub-module was to not do a complete decoding of the `dune-project` file and keep enough information to edit it. However, now that we are fully loading the `dune-project` file it makes less sense to also load it again in a minimal way on the side. At this point we are only relying on this module for editing the `dune-project` file, so I propose to rename it `Dune_project_editor` instead, to state its purpose more clearly.

We could also extend the main `Dune_project` module to keep the necessary information to do the edition. It's not a requirement for this PR, but you are welcome to have a look if you feel motivated :)